config/config.go

```diff
@@ -156,27 +153,59 @@ func (c *Configuration) Unmarshal(v interface{}) error {
 	return nil
 }
 
+var (
+	tagRe    = regexp.MustCompile("((\\w+:\"[^\"]*\")\\b?)+")
+	emptyEnv = EnvironmentOf(MapFetcher(nil))
+)
+
+type lookup struct {
+	key string
+	env Environment
+}
+
+func (l *lookup) Get() (interface{}, bool) { return l.env.Get(l.key) }
+func (l *lookup) Int(or int) int           { return l.env.Int(l.key, or) }
+func (l *lookup) Bool(or bool) bool        { return l.env.Bool(l.key, or) }
+
 // parseTag returns the key, environment, and optional error assosciated with a
 // given tag. It will return the XOR of either the `git` or `os` tag. That is to
 // say, a field tagged with EITHER `git` OR `os` is valid, but pone tagged with
 // both is not.
 //
 // If neither field was found, then a nil environment will be returned.
-func (c *Configuration) parseTag(tag reflect.StructTag) (key string, env Environment, err error) {
-	git, os := tag.Get("git"), tag.Get("os")
+func (c *Configuration) parseTag(tag reflect.StructTag) []*lookup {
```

Comment:
What if this returned `map[string]Environment, error`? Then you could drop the `lookup` type and call `(Environment) Get(key)` or whatever. Then, it could directly pass the error to `Unmarshal()`, instead of calling `panic`.